test/python/test_onedim.py

```diff
@@ -1301,6 +1301,116 @@ def test_restart(self):
         self.assertNear(mdot[0], sim.reactants.mdot, 1e-4)
         self.assertNear(mdot[-1], -sim.products.mdot, 1e-4)
 
+class TestCounterflowPremixedFlameNonIdeal(utilities.CanteraTest):
+    # Note: to re-create the reference file:
+    # (1) set PYTHONPATH to build/python.
+    # (2) Start Python and run:
+    #     >>> import cantera.test
+    #     >>> t = cantera.test.test_onedim.TestCounterflowPremixedFlameNonIdeal()
+    #     >>> t.setUpClass()
+    #     >>> t.test_mixture_averaged(True)
+    # (3) Compare the reference files created in the current working directory with
+    #     the ones in test/data and replace them if needed.
+
+    def test_mixture_averaged(self, saveReference=False):
+        T_in = 373.0  # inlet temperature
+        comp = 'H2:1.6, O2:1, AR:7'  # premixed gas composition
+
+        gas = ct.Solution("h2o2.yaml", "ohmech-RK")
+        gas.TPX = T_in, 0.05 * ct.one_atm, comp
+        width = 0.2 # m
```

Comment:
I think it would be useful to have this regression test at a pressure high enough where the non-ideal model has some effect (if that's even possible for a H2 flame -- I'm not entirely sure).

Reply:
I have modified the test to run at 10 bar, but truly speaking the temperatures are very high. So there is no difference between the idea gas and real gas models.

Reply:
You could go farther away from stoichiometry? At the same time, this is `h2o2.yaml` …)